lib/pharos/host/el7/scripts/configure-cri-o.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/sh
+
+set -e
+
+. /usr/local/share/pharos/util.sh
+
+yum install -y device-mapper-libs
+
+# hack to make cri-o happy
+if [ ! -e /lib64/libdevmapper.so.1.02.1 ]; then
+    ln -s /lib64/libdevmapper.so.1.02 /lib64/libdevmapper.so.1.02.1
+fi
+
+reload_daemon() {
+    if systemctl is-active --quiet crio; then
+        systemctl daemon-reload
+        systemctl restart crio
+    fi
+}
+
+tmpfile=$(mktemp /tmp/crio-service.XXXXXX)
+cat <<"EOF" >${tmpfile}
+[Unit]
+Description=Open Container Initiative Daemon
+Documentation=https://github.com/kubernetes-incubator/cri-o
+After=network-online.target
+
+[Service]
+Type=notify
+Environment=GOTRACEBACK=crash
+ExecStartPre=/sbin/sysctl -w net.ipv4.ip_forward=1
+ExecStart=/usr/local/bin/crio \
+          $CRIO_STORAGE_OPTIONS \
+          $CRIO_NETWORK_OPTIONS
+ExecReload=/bin/kill -s HUP $MAINPID
+TasksMax=infinity
+LimitNOFILE=1048576
+LimitNPROC=1048576
+LimitCORE=infinity
+OOMScoreAdjust=-999
+TimeoutStartSec=0
+Restart=on-abnormal
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+if diff $tmpfile /etc/systemd/system/crio.service > /dev/null ; then
+    cat $tmpfile > /etc/systemd/system/crio.service
+    rm $tmpfile
```

Comment:
The tmpfile still leaks in the `else` case...

```sh
if ...; then
  mv $tmpfile /etc/systemd/system/crio.service
else
  rm $tmpfile
end
```

Reply:
True, btw should we use `install` here?

Reply:
Maybe it does not matter much... fixing with mv & rm.